src/lib/insights/client.ts

```diff
@@ -0,0 +1,99 @@
+import {
+  Hits,
+  InsightsClient,
+  InsightsClientMethod,
+  InsightsClientPayload,
+  Renderer,
+  RenderOptions,
+  Unmounter,
+  WidgetFactory,
+} from '../../types';
+import find from 'lodash/find';
+
+export const inferPayload = (
+  method: InsightsClientMethod,
+  results: RenderOptions['results'],
+  hits: Hits<unknown>,
+  objectIDs: string[]
+): InsightsClientPayload => {
+  const { index, queryID } = results;
+  switch (method) {
+    case 'clickedObjectIDsAfterSearch': {
+      const positions = objectIDs.map(objectID => {
+        const hit = find(hits, hit => hit.objectID === objectID);
+        // NOTE: we might want to optimize this so as we browse though the
+        // hits once, which can get pretty big
+        if (typeof hit === 'undefined') {
+          throw new Error(
+            `Could not find ObjectID ${objectID} in the returned hits.`
+          );
+        }
+        return hit.__position;
+      });
+      return { index, queryID, objectIDs, positions };
+    }
+
+    case 'convertedObjectIDsAfterSearch':
+      return { index, queryID, objectIDs };
+
+    default:
+      throw new Error(`Unsupported method passed ${method}`);
+  }
+};
+
+const _wrapInsightsClient = (
+  aa: InsightsClient,
+  results: RenderOptions['results'],
+  hits: Hits<unknown>
+): InsightsClient => (
+  method: InsightsClientMethod,
+  payload: InsightsClientPayload
+) => {
+  if (typeof aa !== 'function') {
```

Comment:
Isn't this check already performed there?

https://github.com/algolia/instantsearch.js/pull/3598/files#diff-bde03140d12ec15ad28983aea70e6f69R84

Reply:
Can you explain this check before resolving the conversation?

Reply:
it's removed here 20545cd